### PR TITLE
providers: inject current snapcraft snap into instances

### PR DESCRIPTION
**Pull request by mr-cal**
- [X] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [X] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [X] Have you successfully run `make lint`?
- [X] Have you successfully run `pytest tests/unit`?

-----

Allows for core22 snaps to be built using a parallel-installed instance of snapcraft.

A subsequent PR will support core18|20 snaps.

(CRAFT-1660)

### CODE REVIEW

File: snapcraft/providers.py [hunk 1]
```diff
@@ -176,7 +176,19 @@ def get_base_configuration(
     # install snapcraft from the store's stable channel
     snap_channel = get_managed_environment_snap_channel()
     if sys.platform != "linux" and not snap_channel:
+        emit.progress(
+            "Using snapcraft from snap store channel 'latest/stable' in instance "
+            " because snap injection is only supported on Linux hosts.",
```
Comment (tigarmo) [inline snapcraft/providers.py:181]: minor but the full sentence might be double-spaced here

File: tests/spread/core22/parallel-install/task.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  # unset parallel instance support
+  snap set system experimental.parallel-instances=null
+
+  install_lxd
+  install_snapcraft
+
+execute: |
+  #shellcheck source=tests/spread/tools/prepare.sh
+  . "$TOOLS_DIR/prepare.sh"
```
Comment (tigarmo) [inline tests/spread/core22/parallel-install/task.yaml:38]: isn't this being called in `prepare` already?

**Review by tigarmo — CHANGES_REQUESTED** [review]

File: snapcraft/providers.py (hunk 1 above)
Comment (mr-cal) [reply snapcraft/providers.py]: Good catch, thanks!

File: tests/spread/core22/parallel-install/task.yaml (hunk 2 above)
Comment (mr-cal) [reply tests/spread/core22/parallel-install/task.yaml]: Yes, but it has to get re-sourced in every section where I need to call a function from that file.

**Review by tigarmo — APPROVED** [review]

**Pull request merged**